presentation/src/main/java/com/globant/equattrocchio/cleanarchitecture/mvp/presenter/ImagesPresenter.java

```diff
@@ -1,93 +1,124 @@
 package com.globant.equattrocchio.cleanarchitecture.mvp.presenter;
 
 import android.app.Activity;
+import android.view.View;
 
-import com.globant.equattrocchio.cleanarchitecture.util.bus.RxBus;
 import com.globant.equattrocchio.cleanarchitecture.mvp.view.ImagesView;
+import com.globant.equattrocchio.cleanarchitecture.util.bus.RxBus;
 import com.globant.equattrocchio.cleanarchitecture.util.bus.observers.CallServiceButtonObserver;
+import com.globant.equattrocchio.cleanarchitecture.util.bus.observers.CallServiceFABObserver;
+import com.globant.equattrocchio.cleanarchitecture.util.bus.observers.ImageClickObserver;
 import com.globant.equattrocchio.data.ImagesServicesImpl;
 import com.globant.equattrocchio.domain.GetLatestImagesUseCase;
+import com.globant.equattrocchio.domain.response.Image;
+import com.globant.equattrocchio.domain.response.Result;
 
 import io.reactivex.annotations.NonNull;
-import io.reactivex.observers.DefaultObserver;
 import io.reactivex.observers.DisposableObserver;
+import io.realm.Realm;
 
 public class ImagesPresenter {
-
     private ImagesView view;
     private GetLatestImagesUseCase getLatestImagesUseCase;
-
+    private Realm mRealm;
 
     public ImagesPresenter(ImagesView view, GetLatestImagesUseCase getLatestImagesUseCase) {
         this.view = view;
         this.getLatestImagesUseCase = getLatestImagesUseCase;
+        mRealm = Realm.getDefaultInstance();
     }
 
-    public void onCountButtonPressed() {
-
-        view.showText(new String(""));//todo: aca va el string que me devuelva el execute del usecase
-
-
+    public void loadImages(Result result) {
+        view.setProgressVisibility(View.GONE);
+        view.setButtonVisibility(View.GONE);
+        view.loadRecycler(result);
     }
 
-    private void onCallServiceButtonPressed() {
-
-        getLatestImagesUseCase.execute(new DisposableObserver<Boolean>() {
+    public void onCallServiceButtonPressed() {
+        view.setProgressVisibility(View.VISIBLE);
+        getLatestImagesUseCase.execute(new DisposableObserver<Result>() {
             @Override
-            public void onNext(@NonNull Boolean aBoolean) {
-                loadFromPreferences();
+            public void onNext(@NonNull Result result) {
+                loadImages(result);
```

Comment:
Here you have to save the result in Realm, then load the Recycler View from Realm.